### PR TITLE
Draft: test message output from pull

### PR DESCRIPTION
**Pull request by dcarley**
## Proposed Changes

From https://github.com/flox/flox/pull/1337#pullrequestreview-2009466154

**refactor: wip Test handle_pull_result messages**

I've tried using an enum because we need to know the action and content
of the message. We don't have the emoji formatting though because that's
applied later.

Things I'm unsure about:

- using String when the existing methods use a trait?
- not able to do partial matches on the contents
- there are some intermediate warnings that aren't captured

An alternative could be to construct something like a logger that will
accept a different output buffer for the tests. It's a more invasive
change though.

NB: the existing methods would be removed and all callers updated but this just demonstrates the approach.

**chore: wip Additional test for pull errors**

I haven't figured out how to invoke the right branch of the code from
the tests yet.

## Release Notes

N/A

### CODE REVIEW

File: cli/flox/src/commands/pull.rs [hunk 1]
```diff
@@ -290,7 +293,7 @@ impl Pull {
         force: bool,
         mut env: ManagedEnvironment,
         query_functions: Option<QueryFunctions>,
-    ) -> Result<()> {
+    ) -> Result<message::Action, anyhow::Error> {
```
Comment (ysndr) [inline cli/flox/src/commands/pull.rs:296]: `anyhow::Result` defaults to `anyhow::Error` as the error variant.

```suggestion
    ) -> Result<message::Action> {
```

File: cli/flox/src/utils/message.rs [hunk 2]
```diff
@@ -9,6 +9,30 @@ fn print_message(v: impl Display) {
...
+    Plain(String),
+    Error(String),
+    Created(String),
+    Deleted(String),
+    Updated(String),
+    Warning(String),
+}
+
+impl Action {
+    pub fn print(self) {
```
Comment (ysndr) [inline cli/flox/src/utils/message.rs:24]: print can act on a reference alone, no need to move (and drop) the object.

(`v` will be &String, which implements `Display`)

```suggestion
    pub fn print(&self) {
```

File: cli/flox/src/commands/pull.rs [hunk 3]
```diff
@@ -356,12 +357,10 @@ impl Pull {
                             Use 'flox edit' to address issues before activating.
                         "};
 
-                        message::created(message_with_warning);
+                        Ok(message::Action::Warning(message_with_warning))
```
Comment (ysndr) [inline cli/flox/src/commands/pull.rs:360]:
```suggestion
                        Ok(message::Action::Created(message_with_warning))
```

Comment (ysndr) [reply cli/flox/src/commands/pull.rs:360]: i know it says warning, not sure if changing the message type was intentional though

**Review by ysndr — CHANGES_REQUESTED** [review]

**Comment by dcarley** [comment]
We discussed some other options in https://flox-dev.slack.com/archives/C05P6A5J6U8/p1713779412179789 and ultimately decided to punt on this for now but may want to revisit when extending the logic in https://github.com/flox/flox/issues/1226

**Pull request closed without merging**